### PR TITLE
second

### PR DESCRIPTION
**Pull request by StanisLoveSid**
test

### CODE REVIEW

File: library_garage/classes/book.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Book
+	attr_accessor :title, :author
+	def initialize(title, author)
+		@title = title
+		@author = author
+	end
```
Comment (yamnich) [inline library_garage/classes/book.rb]: please, change your editor settings

File: library_garage/db/load_data_to_db.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
+require '../classes/author.rb'
+require '../classes/book.rb'
+require '../classes/reader.rb'
+require '../classes/order.rb'
+require '../classes/library.rb'
+require 'yaml'
+
+class Downloader
```
Comment (yamnich) [inline library_garage/db/load_data_to_db.rb]: 1. Let's try to use module and include it to Library

File: library_garage/db/load_data_to_db.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  end
+  
+  def data_loader
+    @db.books << Book.new('Taras Bulba', 'Nikolai Gogol')
+    @db.books << Book.new('Kobzar', 'Taras Shevchenko')
+    @db.books << Book.new('Ruby programming language', 'Yukihiro Matsumoto')
+    @db.books << Book.new('The Master and Margarita', 'Michael Bulgakov')
+    @db.books << Book.new('The White Guard', 'Michael Bulgakov')
+    @db.books << Book.new('Rich Dad, Poor Dad', 'Robert Kiyosaki')
+    @db.books << Book.new('Sheep Chase', 'Haruki Murakami')
```
Comment (yamnich) [inline library_garage/db/load_data_to_db.rb]: we can use loop here

File: library_garage/db/load_data_to_db.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    @db.orders << Order.new('Ruby programming language', 'User8', '01.01.2016')
+    @db.orders << Order.new('Ruby programming language', 'User8', '01.01.2016')
+    @db.orders << Order.new('Ruby programming language', 'User8', '01.01.2016')
+    @db.orders << Order.new('Ruby programming language', 'User8', '01.01.2016')
+    @db.orders << Order.new('Ruby programming language', 'User8', '01.01.2016')
+    @db.orders << Order.new('Ruby programming language', 'User8', '01.01.2016')
+    @db.orders << Order.new('Ruby programming language', 'User8', '01.01.2016')
+    @db.orders << Order.new('Ruby programming language', 'User8', '01.01.2016')
+
+	save_data_to_db
```
Comment (yamnich) [inline library_garage/db/load_data_to_db.rb]: do not use it here

File: library_garage/exe.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
+require './classes/author.rb'
+require './classes/book.rb'
+require './classes/reader.rb'
+require './classes/order.rb'
+require './classes/library.rb'
+require 'yaml'
+
+
+class Exe
```
Comment (yamnich) [inline library_garage/exe.rb]: rename it please and change to module

File: library_garage/exe.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+  def initialize
+    @database = YAML.load(File.read("./db/database.yaml"))
+  end
+
+  def active_reader
+    r = @database.orders.group_by { |o| o.reader }.sort_by { |x, y| y.count }
```
Comment (yamnich) [inline library_garage/exe.rb]: do not name it with one letter!
you can avoid using local variable here

File: library_garage/exe.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    r.dig(r.size-1,0)
+  end
+
+  def popular_book
+    sorted_books.dig(0,0)
+  end
+
+  def sorted_books
+    @database.orders.group_by { |o| o.book }.sort_by { |x, y| y.count }.reverse
```
Comment (yamnich) [inline library_garage/exe.rb]: please, name variables properly

File: library_garage/exe.rb [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  end
+
+  def top_3_book_users
+  	sorted_books.first(3).collect {|x, y| y }.flatten.collect {|o| o.reader }.uniq.count
```
Comment (yamnich) [inline library_garage/exe.rb]: please, name variables properly

File: library_garage/db/load_data_to_db.rb (hunk 2 above)
Comment (StanisLoveSid) [reply library_garage/db/load_data_to_db.rb]: We are using here 'load_data_to_db.rb' and class Downloader only for filling the database. It is used only once before finding the statistics. It is not used each time.

Comment (StanisLoveSid) [reply library_garage/db/load_data_to_db.rb]: Can it be left as a class?

Comment (yamnich) [reply library_garage/db/load_data_to_db.rb]: We discussed it on review. This class has no own data, it only used library and its data/methods, so you can't simply create library and fill it at any time. Downloader will create new library each time. Module will be better here.

File: library_garage/classes/order.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+class Order
+  attr_accessor :book, :reader, :date
+  def initialize(book, reader, date)
+	@book = book
```
Comment (yamnich) [inline library_garage/classes/order.rb]: настройте редактор пожалуйста

File: library_garage/statistics.rb [hunk 10]
```diff
@@ -0,0 +1,19 @@
+module Statistics
+
+  def active_reader
+    @orders.group_by { |order| order.reader }.sort_by { |reader, order| order.count }.reverse.dig(0,0)
```
Comment (yamnich) [inline library_garage/statistics.rb]: group_by(&:reader)

File: library_garage/statistics.rb [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  def popular_book
+    sorted_books.dig(0,0)
+  end
+
+  def sorted_books
+    @orders.group_by { |order| order.book }.sort_by { |book, order| order.count }.reverse
```
Comment (yamnich) [inline library_garage/statistics.rb]: group_by(&:book)

File: library_garage/statistics.rb [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  def top_3_book_users
+    sorted_books.first(3).collect {|book, order| order }.flatten.collect {|order| order.reader }.uniq.count
```
Comment (yamnich) [inline library_garage/statistics.rb]: collect(&:reader)

**Review by yamnich — COMMENTED** [review]
несколько маленьких замечаний, смержите потом пожалуйста

**Pull request merged**